control-plane/agents/src/bin/core/controller/wrapper.rs

```diff
@@ -1150,13 +1152,23 @@ impl ClientOps for Arc<tokio::sync::RwLock<NodeWrapper>> {
 
     async fn shutdown_nexus(&self, request: &ShutdownNexus) -> Result<(), SvcError> {
         let dataplane = self.grpc_client_locked(request.id()).await?;
-        let result = dataplane.shutdown_nexus(request).await;
-        let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
-        self.update_nexus_states(ctx.deref_mut()).await?;
-        match result {
-            // TODO: Handle idempotency
-            Ok(_) => Ok(()),
-            Err(error) => Err(error),
+        if let Some(nexus) = self.read().await.nexus(&request.uuid()) {
+            if nexus.status == NexusStatus::Shutdown {
+                Ok(())
+            } else {
+                let result = dataplane.shutdown_nexus(request).await;
+                let mut ctx = dataplane.reconnect(GETS_TIMEOUT).await?;
+                self.update_nexus_states(ctx.deref_mut()).await?;
+                match result {
+                    // TODO: Handle idempotency.
+                    Ok(_) => Ok(()),
```

Comment:
idempotency has been implemented: https://github.com/openebs/mayastor/blob/2c181e0abac9ccf6a2b9f424d8281bed363b53d8/io-engine/src/bdev/nexus/nexus_bdev.rs#L654

Reply:
We are already ensuring idempotency in control plane if Nexus is Shutdown. There is one inbetween state (ShuttingDown) which is dataplane specific state. Incase we receive a shutdown request when there is already a request sent by control plane and we are awaiting on that call. That scenario will be handled in saparate PR.

Reply:
I don't think we need anything else as Shutdown is already idempotent from dataplane. We can simply pass-trough the result to the caller.